cmd/kubebuilder/initproject/vendor.go

```diff
@@ -17,172 +17,250 @@ limitations under the License.
 package initproject
 
 import (
-	"archive/tar"
-	"compress/gzip"
 	"fmt"
-	"io/ioutil"
-	"log"
+	"io"
 	"os"
-	"path/filepath"
+	"os/exec"
 
 	"github.com/spf13/cobra"
 
+	"github.com/kubernetes-sigs/kubebuilder/cmd/kubebuilder/util"
 	"github.com/kubernetes-sigs/kubebuilder/cmd/kubebuilder/version"
 )
 
+const (
+	depManifestFile = "Gopkg.toml"
+)
+
 var vendorInstallCmd = &cobra.Command{
 	Use:   "dep",
-	Short: "Install Gopkg.toml, Gopkg.lock and vendor/.",
-	Long:  `Install Gopkg.toml, Gopkg.lock and vendor/.`,
-	Example: `# Bootstrap vendor/ from the src packaged with kubebuilder
-kubebuilder init dep
+	Short: "Install Gopkg.toml and update vendor dependencies.",
+	Long:  `Install Gopkg.toml and update vendor dependencies.`,
+	Example: `Update the vendor dependencies:
+kubebuilder update vendor
 `,
 	Run: RunVendorInstall,
 }
 
 var builderCommit string
 var Update bool
 
-// deleteOld delete all the versions for all packages it is going to untar
-func deleteOld() {
-	// Move up two directories from the location of the `kubebuilder`
-	// executable to find the `vendor` directory we package with our
-	// releases.
-	e, err := os.Executable()
-	if err != nil {
-		log.Fatal("unable to get directory of kubebuilder tools")
-	}
-
-	e = filepath.Dir(filepath.Dir(e))
-
-	// read the file
-	f := filepath.Join(e, "bin", "vendor.tar.gz")
-	fr, err := os.Open(f)
-	if err != nil {
-		log.Fatalf("failed to read vendor tar file %s %v", f, err)
+func RunVendorInstall(cmd *cobra.Command, args []string) {
+	if Update {
+		backupFilename := fmt.Sprintf("%s.bkp", depManifestFile)
+		if err := copyFile(depManifestFile, backupFilename); err != nil {
```

Comment:
Maybe you can use `os.Rename` here. Base on the code change, it is OK to not keep the original file.

Reply:
Done. Thanks for the suggestion.